CMakeLists.txt

```diff
@@ -15,82 +15,131 @@ endif()
+    else()
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32 ")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32 ")
+    endif()
+
+    # -lrt ... see: https://github.com/eclipsesource/J2V8/issues/292
+    set (j2v8_Debug_libs "-lrt")
+    set (j2v8_Release_libs"-lrt")
 #}
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 #{
     set(JAVA_PLATFORM_NAME "darwin")
 
-    # output library filename
-    set(J2V8_LIB_PLATFORM_NAME "macosx")
+    # output library filename parts
     set(J2V8_LIB_PREFIX "")
-    set(J2V8_LIB_ARCH_NAME "x86")
+    set(J2V8_LIB_ARCH_NAME ${J2V8_TARGET_ARCH})
+    set(J2V8_LIB_VENDOR_NAME "")
+    set(J2V8_LIB_PLATFORM_NAME "macosx")
+
+    # configure library architecture
+    if(J2V8_BUILD_X64)
+        set(CMAKE_OSX_ARCHITECTURES "x86_64")
+    else()
+        set(CMAKE_OSX_ARCHITECTURES "i386")
+
+        # fix for 32-bit linking error "ld: illegal text reloc"
+        # see: https://stackoverflow.com/a/9322458/425532
+        set(CMAKE_SHARED_LINKER_FLAGS "-read_only_relocs suppress")
+    endif()
 #}
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 #{
     set(JAVA_PLATFORM_NAME "win32")
 
-    # output library filename
-    set(J2V8_LIB_PLATFORM_NAME "win32")
+    # output library filename parts
     set(J2V8_LIB_PREFIX "lib")
-    set(J2V8_LIB_ARCH_NAME "x86")
+    set(J2V8_LIB_ARCH_NAME ${J2V8_TARGET_ARCH})
+    set(J2V8_LIB_VENDOR_NAME "")
+    set(J2V8_LIB_PLATFORM_NAME "windows")
 #}
 endif()
 
-#-----------------------------------------------------------------------
-# DEPENDENCY SETTINGS / CMAKE OPTIONS
-#-----------------------------------------------------------------------
-
-# look for dependencies
-find_package(Java)
-
-# j2v8 dependency options
-set(J2V8_JDK_DIR ${Java_ROOT} CACHE STRING "Path to the Java JDK dependency")
-set(J2V8_NODEJS_DIR "${CMAKE_SOURCE_DIR}/node" CACHE STRING "Path to the Node.js dependency")
-
-# get the required Node.js link libraries
-get_njs_libs(${J2V8_NODEJS_DIR} "Debug")
-get_njs_libs(${J2V8_NODEJS_DIR} "Release")
-
-# j2v8 build options
-option(J2V8_NODE_COMPATIBLE "Build the J2V8 native bridge with Node.js support enabled" ON)
-option(J2V8_BUILD_ONLY_DEBUG_RELEASE "Generate only Debug and Release configurations (exclude RelWithDebInfo and MinSizeRel)" ON)
-
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows" AND MSVC)
-#{
-    option(J2V8_LINK_WITH_STATIC_MSVCRT "Link against the static version of the Microsoft Visual C++ Common Runtime (will link against the dynamic DLL version if this option is disabled)" ON)
-#}
-endif()
+message("--------------------------------------------------")
+message("J2V8_LIB_ARCH_NAME     = ${J2V8_LIB_ARCH_NAME}")
+message("J2V8_LIB_VENDOR_NAME   = ${J2V8_LIB_VENDOR_NAME}")
+message("J2V8_LIB_PLATFORM_NAME = ${J2V8_LIB_PLATFORM_NAME}")
+message("J2V8_TARGET_ARCH       = ${J2V8_TARGET_ARCH}")
+message("J2V8_BUILD_X64         = ${J2V8_BUILD_X64}")
+message("--------------------------------------------------")
```

Comment:
I would go with a message right at the start. Let's hit those devs with those 2 lines :D